core/controllers/base.py

```diff
@@ -373,7 +373,8 @@ def render_template(self, filepath, iframe_restriction='DENY'):
                 SAMEORIGIN: The template can only be displayed in a frame
                     on the same origin as the page itself.
         """
-        self.response.cache_control.no_cache = True
+
+        self.response.cache_control.no_store = True
```

Comment:
Suggest adding an in-line comment explaining why this is necessary (e.g. the observed issues and why this works around them).

Reply:
As an FYI, this affects back-forward caching. It may cause a performance overhead in Firefox, Opera, Safari and Chrome (Android) users ([reference](https://web.dev/bfcache/#ways-to-opt-out-of-bfcache)). On the other hand, this would actually fix some of the back-forward cache issues we have on the application (examples: https://github.com/oppia/oppia/issues/12587, https://github.com/oppia/oppia/issues/6492).

Reply:
Done.

Reply:
Just noting in this comment that this will probably not actually fix some of the back-forward cache issues Kevin mentioned (see notes later down in the thread).